src/plugins/vis_types/timeseries/public/trigger_action/metrics_helpers.ts

```diff
@@ -0,0 +1,271 @@
+    const aggMap = SUPPORTED_METRICS[aggregation];
+    let additionalFunctionArgs;
+    if (additionalPipelineAggMap.name === 'percentile' && nestedMetaValue) {
+      additionalFunctionArgs = `, percentile=${nestedMetaValue}`;
+    }
+    formula = `${aggMap.name}(${pipelineAgg}(${additionalPipelineAggMap.name}(${
+      additionalSubFunction.field ?? ''
+    }${additionalFunctionArgs ? `${additionalFunctionArgs}` : ''})))`;
+  } else {
+    let additionalFunctionArgs;
+    if (pipelineAgg === 'percentile' && percentileValue) {
+      additionalFunctionArgs = `, percentile=${percentileValue}`;
+    }
+    formula = `${aggregationMap.name}(${pipelineAgg}(${subFunctionMetric.field}${
+      additionalFunctionArgs ? `${additionalFunctionArgs}` : ''
+    }))`;
+  }
+  return formula;
+};
+
+export const getSiblingPipelineSeriesFormula = (
+  aggregation: MetricType,
+  currentMetric: Metric,
+  metrics: Metric[]
+) => {
+  const subFunctionMetric = metrics.find((metric) => metric.id === currentMetric.field);
+  if (!subFunctionMetric) {
+    return null;
+  }
+  const pipelineAggMap = SUPPORTED_METRICS[subFunctionMetric.type];
+  if (!pipelineAggMap) {
+    return null;
+  }
+  const aggregationMap = SUPPORTED_METRICS[aggregation];
+  const subMetricField = subFunctionMetric.field;
+  // support nested aggs with formula
+  const additionalSubFunction = metrics.find((metric) => metric.id === subMetricField);
+  let formula = `${aggregationMap.name}(`;
+  if (additionalSubFunction) {
+    const additionalPipelineAggMap = SUPPORTED_METRICS[additionalSubFunction.type];
+    if (!additionalPipelineAggMap) {
+      return null;
+    }
+    formula += `${pipelineAggMap.name}(${additionalPipelineAggMap.name}(${
+      additionalSubFunction.field ?? ''
+    })))`;
+  } else {
+    formula += `${pipelineAggMap.name}(${subFunctionMetric.field ?? ''}))`;
+  }
+  return formula;
+};
+
+const escapeQuotes = (str: string) => {
+  return str?.replace(/'/g, "\\'");
+};
+
+export const getFilterRatioFormula = (currentMetric: Metric) => {
+  const { numerator, denominator } = currentMetric;
+  return `count(${numerator?.language === 'kuery' ? 'kql' : 'lucene'}='${
+    numerator?.query && typeof numerator?.query === 'string'
+      ? escapeQuotes(numerator?.query)
+      : numerator?.query ?? '*'
+  }') / count(${denominator?.language === 'kuery' ? 'kql' : 'lucene'}='${
+    denominator?.query && typeof denominator?.query === 'string'
+      ? escapeQuotes(denominator?.query)
+      : denominator?.query ?? '*'
+  }')`;
+};
+
+export const getFormulaEquivalent = (
+  currentMetric: Metric,
+  metrics: Metric[],
+  metaValue?: number
+) => {
+  const aggregation = SUPPORTED_METRICS[currentMetric.type]?.name;
+  let formula = null;
+  switch (currentMetric.type) {
+    case 'avg_bucket':
+    case 'max_bucket':
+    case 'min_bucket':
+    case 'sum_bucket': {
+      formula = getSiblingPipelineSeriesFormula(currentMetric.type, currentMetric, metrics);
```

Comment:
nit: as there's no more processing afterworse, all these `formula` thing may just be returned avoiding all these `break`

```suggestion
      return getSiblingPipelineSeriesFormula(currentMetric.type, currentMetric, metrics);
```